src/xbot/common/subsystems/pose/BasePoseSubsystem.java

```diff
@@ -0,0 +1,185 @@
+package xbot.common.subsystems.pose;
+
+import org.apache.log4j.Logger;
+
+import xbot.common.command.BaseSubsystem;
+import xbot.common.command.PeriodicDataSource;
+import xbot.common.controls.sensors.NavImu.ImuType;
+import xbot.common.controls.sensors.XGyro;
+import xbot.common.injection.wpi_factories.WPIFactory;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.XYPair;
+import xbot.common.properties.BooleanProperty;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public abstract class BasePoseSubsystem extends BaseSubsystem implements PeriodicDataSource {
+
+    public final XGyro imu;
+    
+    private final DoubleProperty leftDriveDistance;
+    private final DoubleProperty rightDriveDistance;
+    
+    private final DoubleProperty totalDistanceX;
+    private final DoubleProperty totalDistanceY;
+    
+    private ContiguousHeading currentHeading;
+    private final DoubleProperty currentHeadingProp;
+    
+    private ContiguousHeading lastImuHeading;
+    
+    // These are two common robot starting positions - kept here as convenient shorthand.
+    public static final double FACING_AWAY_FROM_DRIVERS = 90;
+    public static final double FACING_TOWARDS_DRIVERS = -90;
+    
+    private final DoubleProperty currentPitch;
+    private final DoubleProperty currentRoll;
+    
+    private final DoubleProperty inherentRioPitch;
+    private final DoubleProperty inherentRioRoll;
+    
+    private double previousLeftDistance;
+    private double previousRightDistance;
+    
+    private BooleanProperty rioRotated;
+    
+    public BasePoseSubsystem(WPIFactory factory, XPropertyManager propManager) {
+        log.info("Creating");
+        imu = factory.getGyro(ImuType.navX);
+        
+        currentHeadingProp = propManager.createEphemeralProperty("CurrentHeading", 0.0);
+        // Right when the system is initialized, we need to have the old value be
+        // the same as the current value, to avoid any sudden changes later
+        lastImuHeading = imu.getYaw();
+        currentHeading = new ContiguousHeading(FACING_AWAY_FROM_DRIVERS);
+        
+        currentPitch = propManager.createEphemeralProperty("CurrentPitch", 0.0);
+        currentRoll = propManager.createEphemeralProperty("CurrentRoll", 0.0);
+        
+        leftDriveDistance = propManager.createEphemeralProperty("LeftDriveDistance", 0.0);
```

Comment:
We should have a pow wow to talk about property naming scheme. I agree we should strive for consistency but I think we might have different views on what that scheme should be.